tests/suites/test_suite_mpi.function

```diff
@@ -1720,6 +1720,406 @@ exit:
 }
 /* END_CASE */
 
+/* BEGIN_CASE */
+void mpi_core_add_if( char * input_A, char * input_B,
+                      char * input_X4, int carry4,
+                      char * input_X8, int carry8 )
```

Comment:
The test cases give two numbers, and their sum and carry, when working in fixed-length of the larger number (always the second one). The sum and carry can obviously differ depending on whether we have 4-byte or 8-byte MPIs. These are calculated by the code that generates the test cases, which is written in a higher-level language than C. The test code just chooses one of these.

To me, having the test code split out the sum and carry would make the test code more cognisant of MPIs and doing fixed-width arithmetic than it should be - that is definitely closely related to handling carry (something we actually want to test, not to have the test code implement in the same language we're testing). Whereas choosing one set of answers or another, depending on the size of MPI, is very simple.

So again, we disagree significantly on this - I don't think there is "complexity" in having both sets of answers, and I think choosing one set or another is definitely simpler, and it's more obvious that the test code does this correctly, than it would be to have the test code figure out the carry and balance.